### ISSUE TITLE
Panic on reorgOnNewHead

### ISSUE THREAD

**Issue by andrewheadricke** [ethereum/go-ethereum#21505]
Geth 1.9.20 - Linux

```
INFO [08-31|21:47:06.926] Imported new block headers               count=2   elapsed=15.987ms  number=10768509 hash="7fa573…25a30f" ignored=1
INFO [08-31|21:47:07.134] Loaded most recent local header          number=10766460 hash="b45f35…72944b" td=17121215388538268308988 age=7h35m7s
WARN [08-31|21:47:07.135] Rolled back chain segment                header=10768509->10766460 fast=0->0 block=0->0     reason="syncing canceled (requested)"
WARN [08-31|21:47:07.135] Synchronisation failed, retrying         err="no peers to keep download active"
INFO [08-31|21:47:14.766] Looking for peers                        peercount=0 tried=13 static=0
INFO [08-31|21:47:24.770] Looking for peers                        peercount=0 tried=8  static=0
INFO [08-31|21:47:34.774] Looking for peers                        peercount=0 tried=10 static=0
INFO [08-31|21:47:44.778] Looking for peers                        peercount=0 tried=14 static=0
INFO [08-31|21:47:54.781] Looking for peers                        peercount=0 tried=6  static=0
INFO [08-31|21:48:04.785] Looking for peers                        peercount=0 tried=7  static=0
INFO [08-31|21:48:14.788] Looking for peers                        peercount=0 tried=9  static=0
INFO [08-31|21:48:24.792] Looking for peers                        peercount=0 tried=9  static=0
WARN [08-31|21:48:33.698] Synchronisation failed, dropping peer    peer=09b73c91a98e588184b67e95baa15602fa63301c69814c27193a55f48e370a3e err=timeout
INFO [08-31|21:48:34.795] Looking for peers                        peercount=1 tried=12 static=0
INFO [08-31|21:48:41.295] Imported new block headers               count=192 elapsed=829.917ms number=10766652 hash="433ff3…2749ba" age=6h53m
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x1c0 pc=0xd20998]

goroutine 49 [running]:
github.com/ethereum/go-ethereum/light.(*TxPool).reorgOnNewHead(0xc000f5c000, 0x1491740, 0xc002d776e0, 0xc003abb200, 0xc002d776e0, 0xc0047ff020, 0x4664ac)
        github.com/ethereum/go-ethereum/light/txpool.go:229 +0x238
github.com/ethereum/go-ethereum/light.(*TxPool).setNewHead(0xc000f5c000, 0xc003abb200)
        github.com/ethereum/go-ethereum/light/txpool.go:310 +0xf7
github.com/ethereum/go-ethereum/light.(*TxPool).eventLoop(0xc000f5c000)
        github.com/ethereum/go-ethereum/light/txpool.go:291 +0x56
created by github.com/ethereum/go-ethereum/light.NewTxPool
        github.com/ethereum/go-ethereum/light/txpool.go:108 +0x3da
```

**Comment by andrewheadricke** [comment]
I seem to be getting the same issue on 1.9.21

```
INFO [09-12|10:04:18.465] Imported new block headers               count=2    elapsed=13.159ms    number=10843759 hash="be8fd1…1a25b2" ignor
ed=1
INFO [09-12|10:04:18.662] Loaded most recent local header          number=10841710 hash="c34027…6bbc97" td=17342157617971815106040 age=7h24m
6s
WARN [09-12|10:04:18.662] Rolled back chain segment                header=10843759->10841710 fast=0->0 block=0->0     reason="syncing cancel
ed (requested)"
WARN [09-12|10:04:18.662] Synchronisation failed, retrying         err="no peers to keep download active"
INFO [09-12|10:04:22.236] Looking for peers                        peercount=0 tried=15 static=0

... 12 minutes later ...

INFO [09-12|10:16:23.602] Looking for peers                        peercount=1 tried=8  static=0
INFO [09-12|10:16:32.153] Imported new block headers               count=192  elapsed=811.391ms   number=10841902 hash="966147…888e86" age=6h51m55s
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x1c0 pc=0xda5a78]

goroutine 60 [running]:
github.com/ethereum/go-ethereum/light.(*TxPool).reorgOnNewHead(0xc0002bd600, 0x1520660, 0xc00254fb00, 0xc002e64fc0, 0xc00254fb00, 0xc003cb6bc0, 0x4664ac)  
        github.com/ethereum/go-ethereum/light/txpool.go:229 +0x238
github.com/ethereum/go-ethereum/light.(*TxPool).setNewHead(0xc0002bd600, 0xc002e64fc0)
        github.com/ethereum/go-ethereum/light/txpool.go:310 +0xf7
github.com/ethereum/go-ethereum/light.(*TxPool).eventLoop(0xc0002bd600)
        github.com/ethereum/go-ethereum/light/txpool.go:291 +0x56
created by github.com/ethereum/go-ethereum/light.NewTxPool
        github.com/ethereum/go-ethereum/light/txpool.go:108 +0x3da
```

**Comment by karalabe** [comment]
It was fixed on master (unfortunately after the 1.9.21 release). You could
run that (or stick to 1.9.18? or wait one more week until 1.9.22).

**Comment by andrewheadricke** [comment]
Ah ok, no problem. Will wait for next release.
